rpc/src/main/scala/org/bitcoins/rpc/RpcUtil.scala

```diff
@@ -68,25 +78,48 @@ trait RpcUtil extends BitcoinSLogger {
    * @param system An ActorSystem to schedule calls to condition
    */
   def awaitCondition(
-    condition: => Boolean,
+    condition: () => Boolean,
+    duration: FiniteDuration = 100.milliseconds,
+    maxTries: Int = 50,
+    overallTimeout: FiniteDuration = 1.hour)(implicit system: ActorSystem): Unit = {
+    implicit val ec = system.dispatcher
+    def h: Boolean = condition()
```

Comment:
why is this needed? I feel like `condition` is a more descriptive name and should just be used where `h` is

Reply:
This is a bit of type hackery going on here. Basically we need to transform from a 

`() => Boolean`

to 

`() => Future[Boolean]`

the trick is figuring out how to do this WITHOUT caching the value -- but making it so that the `condition` is re-evaluated everytime you call `retryUntilSatisfied`.

Reply:
Got it, a short comment and maybe the name `conditionCallByName` or `conditionDef` wouldn't hurt?